dev_docs/HOW_TO.md

```diff
@@ -0,0 +1,94 @@
+## How to add a new intake field
+
+Adding a new intake field requires changes to both apm-data and apm-server repositories. 
+
+### 1. Make changes in apm-data (this repo)
+
+1. Add the new field to modeldecoder `model.go` such that the field is parsed from JSON.
+   - Intake v2: [input/elasticapm/internal/modeldecoder/v2/model.go](/input/elasticapm/internal/modeldecoder/v2/model.go)
+   - RUM v3: [input/elasticapm/internal/modeldecoder/rumv3/model.go](/input/elasticapm/internal/modeldecoder/rumv3/model.go)
+2. Run `make generate` to generate the corresponding `model_generated.go` from the modified `model.go` in step 1.
+3. Run `make update-licenses` to add license header to `model_generated.go` generated in step 2.
+4. Add the new field to the corresponding file in `model/`, e.g. `span.go`.
+   1. Add the field to the struct.
+   2. Modify the `fields` function to include the field.
+   3. Add a test to the corresponding `*_test.go` file.
+5. Modify the modeldecoder `decoder.go` to map the modeldecoder model in step 1 to the internal model in step 4.
+   - Intake v2: [input/elasticapm/internal/modeldecoder/v2/decoder.go](/input/elasticapm/internal/modeldecoder/v2/decoder.go)
+   - RUM v3: [input/elasticapm/internal/modeldecoder/rumv3/decoder.go](/input/elasticapm/internal/modeldecoder/rumv3/decoder.go)
+6. Run `make test`
+
+### 2. Make changes in [apm-server](https://github.com/elastic/apm-server/)
+
+1. Use the modified apm-data by replacing the apm-data dependency with the local one.
+   - Run
+
+         go mod edit -replace=github.com/elastic/apm-data=/path/to/your/apm-data
+         make update
```

Comment:
```suggestion
         make update
         make
```

nit: you need to run `make` as well, not only `make update`, right?

Reply:
`make` shouldn't be needed for system tests but it is needed if we need the binary. I'll add instructions to test the changes manually.